src/components/SyncView.js

```diff
@@ -51,20 +53,19 @@ export default class SyncView extends React.Component {
     if (!target) return
     this.destroyed = false
     this.stream = replicate.start(target)
-    this.stream.on('data', function (data) {
+    pump(this.stream, through.obj(function (data, enc, next) {
       if (self.destroyed) return
       var row = JSON.parse(data)
       var status = row.topic
       if (status === 'replication-error') return this.onError(new Error(row.message))
       var message = messages[status] || row.message
       self.setState({message, status})
-    })
-    this.stream.on('done', function () {
-      console.log('stream over')
-    })
-    this.stream.on('error', function (err) {
-      console.error(err)
-    })
+      next()
+    }), done)
+
+    function done (err) {
+      if (err) console.error(err)
+    }
   }
 
   componentWillUnmount () {
```

Comment:
What happens if you unmount the view down here and `this.stream === null`?

Reply:
I think that unmounting always happens after component mounts so stream should always exist, but could make sure!

Reply:
But isn't `this.stream` created on `replicate` being called? What if the view mounts and it's not called? Sorry this could be my react ignorance showing.